controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -60,27 +63,32 @@
 
             @Override
             public void connectionDropped() {
+                log.warn("CreateSegment {}/{}/{} Connection dropped", scope, stream, segmentNumber);
                 result.completeExceptionally(
                         new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
             }
 
             @Override
             public void wrongHost(WireCommands.WrongHost wrongHost) {
+                log.warn("CreateSegment {}/{}/{} wrong host", scope, stream, segmentNumber);
                 result.completeExceptionally(new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost));
             }
 
             @Override
             public void segmentAlreadyExists(WireCommands.SegmentAlreadyExists segmentAlreadyExists) {
+                log.info("CreateSegment {}/{}/{} segmentAlreadyExists", scope, stream, segmentNumber);
                 result.complete(true);
             }
 
             @Override
             public void segmentCreated(WireCommands.SegmentCreated segmentCreated) {
+                log.info("CreateSegment {}/{}/{} SegmentCreated", scope, stream, segmentNumber);
                 result.complete(true);
             }
 
             @Override
             public void processingFailure(Exception error) {
+                log.error("CreateSegment {}/{}/{} threw exception", scope, stream, segmentNumber, error);
```

Comment:
I suppose the expectation is that the user will know what to do out of the exception, but if we can give some more specific information to guide whoever reads this message on what to do, it would be better. Typically, an error message indicates some potential action from the user/operator.

Reply:
Typical error message here will be illegal state exception (thats the default exception thrown by FailingReplyProcessor. 

There is nothing user can do. Its more to indicate that there is some error that came that we should typically not expect. 

Or its an error that results in failing of the overall operation.

Reply:
It sounds like this is saying that typically this error message will give no indication of how the operator is supposed to react to it. It is scary to have an error message in the log for which it is not clear what the action should be.

To see if I can make a concrete suggestion, what exactly is the source of this error? I want to understand if it is always recoverable, sometimes recoverable, or never recoverable.

Reply:
All the expected responses are handled explicitly. So if we reach here, its because we got an unexpected response/failure which should typically point to a bug in our code.

Reply:
Ok, you are saying this is a generic error message for unexpected error scenarios that we are supposed to figure out via the exception stack trace.

Reply:
yes